src/librustc_errors/emitter.rs

```diff
@@ -311,7 +311,13 @@ impl EmitterWriter {
         if line.annotations.len() == 1 {
             if let Some(ref ann) = line.annotations.get(0) {
                 if let AnnotationType::MultilineStart(depth) = ann.annotation_type {
-                    if source_string[0..ann.start_col].trim() == "" {
+                    if source_string.chars()
+                                    .enumerate()
+                                    .filter(|&(i, _)| i < ann.start_col)
+                                    .map(|(_, v)|v)
```

Comment:
`.chars().take(ann.start_col)`?

Reply:
Oh right, that's better. Not very familiar with iterator APIs.